src/olympia/git/utils.py

```diff
@@ -206,7 +206,11 @@ def is_extracted(self):
 
     @property
     def is_recent(self):
-        if not self.is_extracted:
+        git_description_path = os.path.join(
+            self.git_repository_path,
+            self.GIT_DESCRIPTION
+        )
+        if not self.is_extracted or not os.path.exists(git_description_path):
             return False
```

Comment:
What if it is recent *and* there's no description file? You mentioned in the issue it's possible that something prevented it from being written.  What's the impact of returning `False` in that scenario?

Reply:
We use this method to determine if we can remove a git repo because of an error or if we shouldn't remove it to prevent an endless loop (think the git repo being deleted over and over again).